Gems/Atom/Feature/Common/Assets/Materials/Types/Debug/DebugVertexStreams.materialtype

```diff
@@ -0,0 +1,46 @@
+{
+    "description": "A simple default base material for Atom testing.",
+    "version": 1,
+    "propertyLayout": {
+        "groups": [
+            {
+                "name": "general",
+                "displayName": "General Settings",
+                "description": "General settings."
+            }
+        ],
+        "properties": {
+            "general": [
+                {
+                    "name": "debugVertexStream",
+                    "displayName": "Display Vertex Stream",
+                    "description": "Display vertex stream interpolated over the triangle.",
+                    "type": "Enum",
+                    "defaultValue": "Normals",
+                    "enumValues": [ "Normals", "Tangents", "Bitangents", "UVs", "TangentW" ],
+                    "connection": {
+                        "type": "ShaderOption",
+                        "name": "o_debugVertexStream"
+                    }
+                },
+                {
+                    "name": "reconstructionMethod",
+                    "displayName": "ReconstructionMethod",
+                    "description": "Use the data from the vertices for all streams, use the Surface Gradient technique for calculating both tangents and bitangents, or use the cross product to reconstruct the bitangents.",
+                    "type": "Enum",
+                    "defaultValue": "UseVertexData",
+                    "enumValues": [ "UseVertexData", "UseSurfaceGradient", "ReconstructBitangent" ],
+                    "connection": {
+                        "type": "ShaderOption",
+                        "name": "o_reconstructionMethod"
+                    }
+                }
```

Comment:
Can you add a lua functor that shows reconstructionMethod only debugVertexStream is Tangent or Bitangent?

Reply:
I think that we should keep it very simple and not add indirect files activation so that users can easily go and change it, but it's only an opinion

Reply:
The functor does add some complexity, but nothing outrageous, and it makes it a lot easier to use and know what options have a real impact, by only seeing the options that are applicable to the vertex stream you are displaying.

I tried to accomplish this with the shader option names (e.g., ReconstructBitangent has no impact on Tangents), but it's still not super clear in the UI without adding the functor. It has even more of an impact with uvs, where using a second uv set makes it so only the surface gradient option is supported, and now having the functors lets me hide the other options.